src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceController.cs

```diff
@@ -19,13 +19,14 @@ namespace System.ServiceProcess
     /// and manipulate it or get information about it.
     public class ServiceController : Component
     {
-        private string _machineName;
+        private string _machineName = DefaultMachineName; // Never null, always a valid name
```

Comment:
Since the `_machineName` is set in all but one constructor - it may be better to just fix the one constructor that is not setting it to set it.